core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -34,10 +38,28 @@ oppia.factory('UrlService', ['$window', function($window) {
       return urlParts[1] === 'embed';
     },
     getPathname: function() {
-      return $window.location.pathname;
+      return this.getCurrentUrl().pathname;
+    },
+    getParamValuesAsList: function(queryTag) {
+      var url = this.getCurrentUrl().href.slice(
+        this.getCurrentUrl().href.indexOf('?') + 1).split('&');
+      var parentIdArray = [];
```

Comment:
Why are we talking about parentIds here? This function should not know anything about parent IDs, it should just generically be retrieving values of fields.

fieldValues seems better.

Reply:
Done